src/Application/Models/Broker.js

```diff
@@ -90,6 +90,16 @@ class Broker {
 
     return this._insertIntoNewBroker(newAsset, index);
   }
+
+  updateRevenue(timeDifferenceInSeconds) {
+    const newAssets = [];
+    this._assets.forEach(function (asset) {
+      const income = asset.income * timeDifferenceInSeconds;
+      newAssets.push(asset.addRevenue(income));
+    });
```

Comment:
if everything in `this._assets` gets converted and pushed to `newAssets`, as is the case here, use `this._assets.map`